### DISCUSSION TITLE
Cache & proxy requests to brands.home-assistant.io

### DISCUSSION THREAD

**Discussion by Zocker1999NET** [home-assistant/frontend#12577] [Other feature requests] [answered]
Hi, in reference to #8379, where more than one third-party request domain was listed, I would still propose to cache & proxy brands.home-assistant.io in each Home Assistant instance itself.
It would be even better, if those brand icons could be integrated in Home-Assistant itself so those would be available locally directly. However this might be not possible due to copyright. However implementing to cache those brand icons should be legally possible and could enable users to see those icons when no Internet is available (and they were requested before) and it could reduce traffic to third-party services from the clients as well.
For users with a limited plan at home it will also reduce the data usage on such plan.

If possible, but fully optional such cache could be implemented in a intelligent way based on facts a browser cache cannot use, e.g. prefetch icons of all available integrations at launch and keep the icons cached until Home-Assistant receives an update.

**Comment by frenck** [comment, marked as answer]
This is not a frontend feature request.

Please use our community forums to request larger features that require more than just frontend changes.

Thanks! 👍

**Comment by steverep** [comment]
@Zocker1999NET there are now 2 open issues for better caching brand images:

- #17560 
- https://github.com/home-assistant/brands/issues/4718

Pull requests are open to fix both.

**Comment by steverep** [comment]
Closing this as both PRs I mentioned have been merged and released.
